File: backups/capacitor-sora-editor-1.3.13/android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/lang/EmptyLanguage.java

package io.github.abc15018045126.sora.lang;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import io.github.abc15018045126.sora.lang.analysis.AnalyzeManager;
import io.github.abc15018045126.sora.lang.completion.CompletionPublisher;
import io.github.abc15018045126.sora.lang.format.Formatter;
import io.github.abc15018045126.sora.lang.smartEnter.NewlineHandler;
import io.github.abc15018045126.sora.lang.util.BaseAnalyzeManager;
import io.github.abc15018045126.sora.text.CharPosition;
import io.github.abc15018045126.sora.text.Content;
import io.github.abc15018045126.sora.text.ContentReference;
import io.github.abc15018045126.sora.text.TextRange;
import io.github.abc15018045126.sora.widget.SymbolPairMatch;

/**
 * Empty language
 *
 * @author abc15018045126
 */
public class EmptyLanguage implements Language {


    public final static SymbolPairMatch EMPTY_SYMBOL_PAIRS = new SymbolPairMatch();

    @NonNull
    @Override
    public Formatter getFormatter() {
        return EmptyFormatter.INSTANCE;
    }

    @Override
    public SymbolPairMatch getSymbolPairs() {
        return EMPTY_SYMBOL_PAIRS;
    }

    @Override
    public void requireAutoComplete(@NonNull ContentReference content, @NonNull CharPosition position, @NonNull CompletionPublisher publisher, @NonNull Bundle extraArguments) {

    }

    @Override
    public int getInterruptionLevel() {
        return INTERRUPTION_LEVEL_STRONG;
    }

    @Override
    public NewlineHandler[] getNewlineHandlers() {
        return new NewlineHandler[0];
    }

    @NonNull
    @Override
    public AnalyzeManager getAnalyzeManager() {
        return EmptyAnalyzeManager.INSTANCE;
    }

    @Override
    public int getIndentAdvance(@NonNull ContentReference content, int line, int column) {
        return 0;
    }

    @Nullable
    @Override
    public QuickQuoteHandler getQuickQuoteHandler() {
        return null;
    }


    @Override
    public void destroy() {

    }

    @Override
    public boolean useTab() {
        return false;
    }

    public static class EmptyFormatter implements Formatter {

        public final static EmptyFormatter INSTANCE = new EmptyFormatter();

        @Override
        public void format(@NonNull Content text, @NonNull TextRange cursorRange) {

        }

        @Override
        public void formatRegion(@NonNull Content text, @NonNull TextRange rangeToFormat, @NonNull TextRange cursorRange) {

        }

        @Override
        public void setReceiver(@Nullable FormatResultReceiver receiver) {

        }

        @Override
        public boolean isRunning() {
            return false;
        }

        @Override
        public void destroy() {

        }
    }

    public static class EmptyAnalyzeManager extends BaseAnalyzeManager {

        public final static EmptyAnalyzeManager INSTANCE = new EmptyAnalyzeManager();


        @Override
        public void insert(@NonNull CharPosition start, @NonNull CharPosition end, @NonNull CharSequence insertedContent) {

        }

        @Override
        public void delete(@NonNull CharPosition start, @NonNull CharPosition end, @NonNull CharSequence deletedContent) {

        }

        @Override
        public void rerun() {

        }

    }
}

File: capacitor-sora-editor/android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/lang/EmptyLanguage.kt
package io.github.abc15018045126.sora.lang

import android.os.Bundle
import io.github.abc15018045126.sora.lang.analysis.AnalyzeManager
import io.github.abc15018045126.sora.lang.completion.CompletionPublisher
import io.github.abc15018045126.sora.lang.format.Formatter
import io.github.abc15018045126.sora.lang.smartEnter.NewlineHandler
import io.github.abc15018045126.sora.lang.util.BaseAnalyzeManager
import io.github.abc15018045126.sora.text.CharPosition
import io.github.abc15018045126.sora.text.Content
import io.github.abc15018045126.sora.text.ContentReference
import io.github.abc15018045126.sora.text.TextRange
import io.github.abc15018045126.sora.widget.SymbolPairMatch

/**
 * Empty language
 *
 * @author abc15018045126
 */
open class EmptyLanguage : Language {

    override val formatter: Formatter
        get() = EmptyFormatter.INSTANCE

    override val symbolPairs: SymbolPairMatch
        get() = EMPTY_SYMBOL_PAIRS

    override fun requireAutoComplete(
        content: ContentReference,
        position: CharPosition,
        publisher: CompletionPublisher,
        extraArguments: Bundle
    ) {
    }

    override val interruptionLevel: Int
        get() = Language.INTERRUPTION_LEVEL_STRONG

    override val newlineHandlers: Array<NewlineHandler>?
        get() = emptyArray()

    override val analyzeManager: AnalyzeManager
        get() = EmptyAnalyzeManager.INSTANCE

    override fun getIndentAdvance(content: ContentReference, line: Int, column: Int): Int {
        return 0
    }

    override val quickQuoteHandler: QuickQuoteHandler?
        get() = null

    override fun destroy() {}

    override fun useTab(): Boolean {
        return false
    }

    class EmptyFormatter : Formatter {
        override fun format(text: Content, cursorRange: TextRange) {}

        override fun formatRegion(text: Content, rangeToFormat: TextRange, cursorRange: TextRange) {}

        override fun setReceiver(receiver: Formatter.FormatResultReceiver?) {}

        override fun isRunning(): Boolean {
            return false
        }

        override fun destroy() {}

        companion object {
            @JvmField
            val INSTANCE = EmptyFormatter()
        }
    }

    class EmptyAnalyzeManager : BaseAnalyzeManager() {
        override fun insert(start: CharPosition, end: CharPosition, insertedContent: CharSequence) {}

        override fun delete(start: CharPosition, end: CharPosition, deletedContent: CharSequence) {}

        override fun rerun() {}

        companion object {
            @JvmField
            val INSTANCE = EmptyAnalyzeManager()
        }
    }

    companion object {
        @JvmField
        val EMPTY_SYMBOL_PAIRS = SymbolPairMatch()
    }
}
